qiita_db/metadata_template/prep_template.py

```diff
@@ -75,6 +75,7 @@ class PrepTemplate(MetadataTemplate):
     _sample_cls = PrepSample
     _fp_id = convert_to_id("prep_template", "filepath_type")
     _filepath_table = 'prep_template_filepath'
+    _columns = PREP_TEMPLATE_COLUMNS
```

Comment:
This is not correct. In case that the prep template is of a target gene datatype, additional columns are requested. My suggestion is, instead of using a class variable as done here, create a function call `get_column_restrictions` that each subclass (`SampleTemplate` and `PrepTemplate`) implements. The `SampleTemplate` implementation will return the `SAMPLE_TEMPLATE_COLUMNS` while the `PrepTemplate` implementation will return a new dictionary with the contents of `PREP_TEMPLATE_COLUMNS` + `PREP_TEMPLATE_COLUMNS_TARGET_GENE` in case of being of a `TARGET_GENE_DATA_TYPE`, or `PREP_TEMPLATE_COLUMNS` otherwise. Does this make sense? There is an example of this in the create method of the prep template.